src/lib/supabase/server/index.ts

```diff
@@ -1,13 +1,24 @@
-import { ServerClient } from './createServerClient'
+import { cookies } from "next/headers";
+import { ServerClient, createServerClient } from "./createServerClient";
 
-export * from './createServerClient'
-export * from './createMiddlewareClient'
+export * from "./createServerClient";
+export * from "./createMiddlewareClient";
 
 export async function getUser(supabase: ServerClient) {
-  const { data: { user }, error } = await supabase.auth.getUser()
+  const {
+    data: { user },
+    error,
+  } = await supabase.auth.getUser();
   if (!user || error) {
-    console.error('Error retrieving user:', error)
-    return
+    console.error("Error retrieving user:", error);
+    return;
   }
-  return user
-}
\ No newline at end of file
+  return user;
+}
+
+export const useSupabase = () => {
+  const cookieStore = cookies();
+  const supabase = createServerClient(cookieStore);
+
+  return supabase;
+};
```

Comment:
hooks should be defined in the `hooks` directory
- actual: hook is defined in the `lib` directory
- expected: hooks are defined in `hook` directory